client/my-sites/site-settings/podcast-cover-image-setting/index.jsx

```diff
@@ -0,0 +1,318 @@
+				this.props.deleteMedia( siteId, transientMediaId );
+			} else {
+				this.props.receiveMedia( siteId, media );
+			}
+
+			if ( isUploadInProgress ) {
+				return;
+			}
+
+			MediaStore.off( 'change', checkUploadComplete );
+
+			if ( ! isFailedUpload ) {
+				debug( 'upload media', media );
+				this.props.onSelect( media.ID, media.URL );
+			}
+
+			// Remove transient image so that new image shows or if failed upload, the prior image
+			this.setState( { transientMediaId: null } );
+		};
+
+		MediaStore.on( 'change', checkUploadComplete );
+
+		MediaActions.add( site, {
+			ID: transientMediaId,
+			fileContents: blob,
+			fileName,
+		} );
+	}
+
+	setCoverImage = ( error, blob ) => {
+		if ( error || ! blob ) {
+			return;
+		}
+
+		const { siteId } = this.props;
+		const selectedItem = head( MediaLibrarySelectedStore.getAll( siteId ) );
+		if ( ! selectedItem ) {
+			return;
+		}
+
+		debug( 'selectedItem', selectedItem );
+
+		const { crop, transform } = this.props;
+		const isImageEdited = ! isEqual(
+			{
+				...crop,
+				...transform,
+			},
+			{
+				topRatio: 0,
+				leftRatio: 0,
+				widthRatio: 1,
+				heightRatio: 1,
+				degrees: 0,
+				scaleX: 1,
+				scaleY: 1,
+			}
+		);
+		debug( 'isImageEdited', isImageEdited, { crop, transform } );
+
+		if ( isImageEdited ) {
+			this.uploadCoverImage( blob, `cropped-${ selectedItem.file }` );
+		} else {
+			this.props.onSelect( selectedItem.ID, selectedItem.URL );
+		}
+
+		this.hideModal();
+		this.props.resetAllImageEditorState();
+	};
+
+	cancelEditingCoverImage = () => {
+		this.props.onCancelEditingCoverImage();
+		this.props.resetAllImageEditorState();
+		this.setState( { isEditingCoverImage: false } );
+	};
+
+	isParentReady( selectedMedia ) {
+		return ! selectedMedia.some( item => item.external );
+	}
+
+	preloadModal() {
+		asyncRequire( 'post-editor/media-modal' );
+	}
+
+	renderChangeButton() {
+		const { coverImageId, coverImageUrl, translate } = this.props;
+		const isCoverSet = coverImageId || coverImageUrl;
+
+		return (
+			<Button
+				className="podcast-cover-image-setting__button"
+				compact
+				onClick={ this.showModal }
+				onMouseEnter={ this.preloadModal }
+				type="submit"
+			>
+				{ isCoverSet ? translate( 'Change' ) : translate( 'Add' ) }
+			</Button>
+		);
+	}
+
+	renderCoverPreview() {
+		const { coverImageUrl, siteId, translate } = this.props;
+		const { transientMediaId } = this.state;
+		const media = transientMediaId && MediaStore.get( siteId, transientMediaId );
+		const imageUrl = ( media && media.URL ) || coverImageUrl;
+		const imageSrc = imageUrl && resizeImageUrl( imageUrl, 96 );
+		const isTransient = !! transientMediaId;
+
+		const classNames = classnames( 'podcast-cover-image-setting__preview', {
+			'is-blank': ! imageSrc,
+			'is-transient': isTransient,
+		} );
+
+		return (
+			<button
+				className={ classNames }
+				onClick={ this.showModal }
+				onMouseEnter={ this.preloadModal }
+				type="submit"
```

Comment:
It is very important that all buttons here have `type=submit`.  Should we note this in the code?

Reply:
I'm glad this is no longer needed, as of 1bd9b99